SimplifiedTests/NYPLAxisTests/NYPLAxisServiceTests.swift

```
@@ -0,0 +1,182 @@
+      downloadSucceededExpectation.fulfill()
+    }
+    
+    axisService.fulfillAxisLicense(downloadTask: task)
+    
+    wait(
+      for: [downloadFailedExpectation, downloadSucceededExpectation],
+      timeout: 10.0)
+  }
+  
+  func testAxisServiceShouldFailUponPackageDownloadFailure() {
+    let downloadFailedExpectation = XCTestExpectation(
+      description: "Failed package content download should result in failed download!")
+    let downloadSucceededExpectation = XCTestExpectation(
+      description: "Download shouldn't succeed upon package content download failure")
+    
+    downloadSucceededExpectation.isInverted = true
+    
+    let axisService = createAxisService(licenseService: succeedingLicenseService,
+                                        metadataService: succeedingMetadataService,
+                                        packageService: failingPackageService)
+    
+    let task = URLSessionDownloadTask()
+    
+    downloadBroadcaster.downloadFailed = {
+      downloadFailedExpectation.fulfill()
+    }
+    
+    downloadBroadcaster.downloadSuccessful = {
+      downloadSucceededExpectation.fulfill()
+    }
+    
+    axisService.fulfillAxisLicense(downloadTask: task)
+    
+    wait(
+      for: [downloadFailedExpectation, downloadSucceededExpectation],
+      timeout: 10.0)
+  }
+  
+  func testAxisServiceShouldSucceedWhenAllItemsAreDownloaded() {
+    let downloadSucceededExpectation = XCTestExpectation(
+      description: "Axis Service should succeed upon successful download of all content")
+    
+    let axisService = createAxisService(licenseService: succeedingLicenseService,
+                                        metadataService: succeedingMetadataService,
+                                        packageService: succeedingPackageService)
+    
+    let task = URLSessionDownloadTask()
+    
+    downloadBroadcaster.downloadSuccessful = {
+      downloadSucceededExpectation.fulfill()
+    }
+    
```

can we have a  `downloadFailed = { XCTFail() }` so we can fail faster?

---

Dine

---

I don't see the downloadFailed callback anymore, maybe it got lost while rebasing?

---

rebases 😬

---

I see it now, looks good 👍